File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        for (int i = 0; i < nums.size() - 1; i++) {
            for (int j = i + 1; j < nums.size(); j++) {
                if (nums[i] + nums[j] == target) {
                    vector<int> res;
                    res.push_back(i);
                    res.push_back(j);
                    return res;
                }
            }
        }
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []
        subset = []

        def dfs(i):
            if sum(subset) == target and subset not in res:
                res.append(subset.copy())
                return res
            elif sum(subset) > target or i >= len(nums):
                return
            
            subset.append(nums[i])
            dfs(i)
            dfs(i + 1)

            subset.pop()
            dfs(i + 1)
        
        dfs(0)
        print(res)
        return res